Objects/longobject.c

```diff
@@ -4366,23 +4371,65 @@ long_pow(PyObject *v, PyObject *w, PyObject *x)
         }
     }
     else {
-        /* Left-to-right 5-ary exponentiation (HAC Algorithm 14.82) */
-        Py_INCREF(z);           /* still holds 1L */
-        table[0] = z;
-        for (i = 1; i < 32; ++i)
-            MULT(table[i-1], a, table[i]);
+        /* Left-to-right k-ary sliding window exponentiation
+         * (HAC Algorithm 14.85)
```

Comment:
Spell-out "Handbook of Applied Cryptography".  Perhaps add a link: https://cacr.uwaterloo.ca/hac/about/chap14.pdf

Reply:
There's already a link in the `x_mul()` comments, and yet another just above in the "left-to-right binary exponentiation" block, but I don't think they help. It's impossible not find the book at once via Google, but links can change. I'll spell out the acronym, though.